src/rgw/rgw_sal_motr.cc

```diff
@@ -107,15 +107,16 @@ namespace rgw::sal {
     return 0;
   }
 
-  int MotrUser::load_user_from_motr_idx(const DoutPrefixProvider *dpp,
-		                        RGWUserInfo& info, map<string, bufferlist> *attrs,
-                                        RGWObjVersionTracker *objv_tracker)
+  static int load_user_from_idx(const DoutPrefixProvider *dpp,
```

Comment:
This function exists only because `User::load_user()` does not allow to store the fetched `RGWUserInfo` in some temporary variable (see how it's used at `User::store_user()`). Having another function that does the same thing as `load_user()` in the public API would be confusing, it seems.

I wish `load_user()` would allow saving the fetched `RGWUserInfo` in an optional argument. If the argument is not specified, it would work as it is now (i.e. saving the fetched user info into User::info field).

Reply:
Because there is already such a member function - `::load_user()`. It just doesn't return the needed values via parameters, which we need to check at `::store_user()`. Having another such function in the public API would be confusing, it seems, and would make excess noise. Also, we hide the implementation details this way.